devicemodel/core/mptbl.c

```diff
@@ -117,11 +118,39 @@ static void
 mpt_build_proc_entries(proc_entry_ptr mpep, int ncpu)
 {
 	int i;
+	uint64_t guest_pcpu_bitmask = vm_get_cpu_affinity_dm();
+
+	if (guest_pcpu_bitmask == 0) {
+		pr_err("%s,Err: Invalid guest_pcpu_bitmask.\n", __func__);
+		return;
+	}
+
+	pr_info("%s, guest_cpu_bitmask: 0x%x\n", __func__, guest_pcpu_bitmask);
 
 	for (i = 0; i < ncpu; i++) {
+		int pcpu_id = pcpuid_from_vcpuid(guest_pcpu_bitmask, i);
+		int lapic_id;
+
```

Comment:
Do we need to check whether the pcpu_id is valid ? Now the lapicid_from_pcpuid will also does this check.

Reply:
Hi,lifei,
yes, this patch does including the pcpu_id validity checking, see below code in this patch:
if (pcpu_id < 0) {
			pr_err("%s,Err: pcpu id is not found in guest_pcpu_bitmask.\n", __func__);
			return;
		}

		assert(pcpu_id < ACRN_PLATFORM_LAPIC_IDS_MAX);
		if (pcpu_id >= ACRN_PLATFORM_LAPIC_IDS_MAX) {
			pr_err("%s,Err: pcpu id %u should be less than ACRN_PLATFORM_LAPIC_IDS_MAX.\n", __func__, pcpu_id);
			return;
		}

Reply:
@lifeix  can you double check my patch? it is already checking pcpu_id